### ISSUE TITLE
FlowRouter.getParam returns undefined after redirect (Redirect works properly).

### ISSUE THREAD

**Issue by Shalimov** [kadirahq/flow-router#351]
I have the following case:
I want to prevent user from visiting page if some condition gives true, and I use trigger to prevent this action.
So after redirect I try to use FlowRouter.getParam('year') but it returns me `undefined` in case that I have valid url.
Redirects works properly but I cannot get current params via FlowRouter.getParam.

**Comment by arunoda** [comment]
Yep. This is a valid case and I can see why this is happening. There are weird places in FR due to the subscriptions-integration. If you can give me a sample repo I can work on this. 

If this is fixable, I'll fix it. If not we need to wait until 3.0 (Anyway, I'll suggest a workaround for sure)

**Comment by Shalimov** [comment]
ok, I'll try to provide you example by the end of this day (or tomorrow)

**Comment by Shalimov** [comment]
I've created example:
You can look at the source code here: https://github.com/Shalimov/flow-router-params-test
You can try it on this url: http://frgetparams.meteor.com/

Use Case: 
I've created handler that check the following points:
 - if you select current or past (year, month or day) it gives you ability to visit next page.
 - if you click on (year, month or day) in future it redirects you back on the current url, and prevents visiting of the next page.

On month and day pages it shows FlowRouter params, as you will see after redirect this params will be disappeared

**Comment by johanbrook** [comment]
I also experience this after a redirect. My case is:

1. We have these routes (including redirects):

   ```
/:slug -redirects-to-> /:slug/inbox
/:slug/:projectSlug # renders template
```

2. When visiting `/:slug` via a link (i.e. not full page reload), `FlowRouters.getParam('slug')` returns `undefined`.

**Comment by krizka** [comment]
Is It related? Made simple repo.
https://github.com/krizka/flow-rerender

**Comment by arunoda** [comment]
Thanks we had a pretty big bug and already fixed in the `ssr` branch. @krizka Could you test this with `kadira:flow-router-ssr`

**Comment by krizka** [comment]
Yeah. But It's with kadira:flow-router-ssr version 3.9. Do you have newer?

**Comment by arunoda** [comment]
Okay. I got it. I'll have a look at your repo and will update on this issue: https://github.com/kadirahq/flow-router/issues/486

**Comment by abecks** [comment]
I am having this exact issue with the following code:

```
FlowRouter..route('/companies/:companyId', {
  name: 'companyView',
  template: 'Companies_view_page',
  triggersEnter: [
    function (context, redirect) {
      redirect('companyViewPathways', context.params);
    },
  ],
}));
```

```
Template.Companies_view_page.helpers({
  companyId() {
    console.log(FlowRouter.getParam('companyId');
    return FlowRouter.getParam('companyId');
  },
});
```

Console output after navigating to the route that serves the redirect:
```
ctATDPnyNqHWftyG7
undefined
```

It seems that the correct param is passed through, but then getParam causes another invalidation and re-runs the helper. This time with a value of undefined.

**Comment by pcorey** [comment]
I'm getting this issue as well. @johanbrook's description matches my scenario. After digging into the FR code, it looks like this piece of code that "closes" the oldest route is actually closing the current route:

https://github.com/kadirahq/flow-router/blob/56c5e7e8d025affd41d8532524470fe9737fa189/client/router.js#L459-L461

    if(oldestRoute) {
      oldestRoute.registerRouteClose();
    }

I threw in a quick fix that seems to work, but I don't even begin to understand the real cause of this, or if this fix has any unintended consequences:

    if(oldestRoute && oldestRoute !== currentContext.route) {
      oldestRoute.registerRouteClose();
    }

Any ideas @arunoda?
